.pre-commit-config.yaml

```
@@ -0,0 +1,29 @@
+repos:
+  - repo: https://github.com/charliermarsh/ruff-pre-commit
+    rev: v0.0.292
+    hooks:
+      - id: ruff
+        args: [--fix, --exit-non-zero-on-fix]
+
+  - repo: https://github.com/psf/black
+    rev: 23.9.1
+    hooks:
+      - id: black
+        exclude: 'tests/data/.*'
+
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.4.0
+    hooks:
+      - id: check-toml
+      - id: check-yaml
+      - id: check-merge-conflict
+      - id: end-of-file-fixer
+        exclude: 'tests/data/.*'
+      - id: mixed-line-ending
+      - id: check-added-large-files
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.5.1
+    hooks:
+      - id: mypy
+        args: ['--ignore-missing', '--python-version', '3.8']
```

Ooo, I just hit 'approve' then noticed this. This should probably by 3.10

```suggestion
        args: ['--ignore-missing', '--python-version', '3.10']
```